### PR TITLE
Mech stomp

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Mechs can now crush mobs under foot.
Currently only works on human sized or lower. Hooman has to be already on the floor.

https://github.com/tgstation/TerraGov-Marine-Corps/assets/7869430/bc3f4009-cec2-4fa1-9784-417007e869e7
## Why It's Good For The Game
Mechs are now 200% more enjoyable.
Adds more DYNAMIC PLAYER INTERACTION when mechs and little people are working together, or against each other.
## Changelog
:cl:
add: Mechs will now crush human sized mobs underfoot
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Is there a way not to crush someone like turning on help intent or turning on safety, because im afraid you're just going to get lynched if you do this multiple times by accident.

**Comment by zzurge** [comment]
> Is there a way not to crush someone like turning on help intent or turning on safety, because im afraid you're just going to get lynched if you do this multiple times by accident.

Unfortunately if were ever gonna see this thing deployed consistently again the risk of having a giant mech stomping teammates is going to have to become an occupational hazard,moreover is it really that hard to pay attention to the giant metal machine?

**Comment by Helg2** [comment]
> Unfortunately if were ever gonna see this thing deployed consistently again the risk of having a giant mech stomping teammates is going to have to become an occupational hazard,moreover is it really that hard to pay attention to the giant metal machine?

Well, game is like 50% text based and like half of all the players can't even read properly, so I'm afraid it will be.

**Comment by Arturlang** [comment]
It should be a crunch, not a squish

File: code/modules/vehicles/mecha/_mecha.dm [hunk 1]
```diff
@@ -300,6 +300,32 @@
...
+///Crushing the mob underfoot
+/obj/vehicle/sealed/mecha/proc/run_over(mob/living/crushed)
+	playsound(src, 'sound/effects/splat.ogg', 50, TRUE)
+	add_mob_blood(crushed)
+	var/turf/below_us = get_turf(src)
+	below_us.add_mob_blood(crushed)
+
+	if(crushed.stat == DEAD)
+		return
+	log_combat(src, crushed, "stomed", addition = "(DAMTYPE: [uppertext(BRUTE)])")
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/_mecha.dm:321]:
```suggestion
	log_combat(src, crushed, "stomped on", addition = "(DAMTYPE: [uppertext(BRUTE)])")
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**